test/Marketplace.test.js

```diff
@@ -490,52 +490,53 @@ describe("Marketplace", function () {
   describe("paying out a slot", function () {
     beforeEach(async function () {
       switchAccount(client)
-      await token.approve(marketplace.address, price(request))
+      await token.approve(marketplace.address, maxPrice(request))
       await marketplace.requestStorage(request)
       switchAccount(host)
       await token.approve(marketplace.address, request.ask.collateral)
     })
 
-    it("pays the host when contract has finished and returns collateral", async function () {
-      await waitUntilStarted(marketplace, request, proof, token)
-      await waitUntilFinished(marketplace, requestId(request))
-      const startBalance = await token.balanceOf(host.address)
-      await marketplace.freeSlot(slotId(slot))
-      const endBalance = await token.balanceOf(host.address)
-      expect(endBalance - startBalance).to.equal(
-        pricePerSlot(request) + request.ask.collateral
-      )
-    })
+    it("pays the host when contract has finished for time he hosted the request and returns collateral to host collateral address", async function () {
+      await advanceTimeForNextBlock(request.expiry / 2)
+      await marketplace.fillSlot(requestId(request), slot.index, proof)
+      const startTime = await currentTime()
 
-    it("pays to host reward address when contract has finished and returns collateral to host collateral address", async function () {
-      await waitUntilStarted(marketplace, request, proof, token)
+      await waitUntilStarted(marketplace, request, proof, token, [slot.index])
```

Comment:
🟠 At a cursory glance, I'd expect this statement to wait until the slot with `slot.index` is started. Instead it does the opposite. You can solve this by making it a named parameter:

```suggestion
      await waitUntilStarted(marketplace, request, proof, token, { ignoreSlotIndexes: [slot.index] })
```